src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -184,6 +217,65 @@ int CPUPerformanceInterface::CPUPerformance::cpu_loads_process(double* pjvmUserL
   _jvm_user_nanos = jvm_user_nanos;
   _jvm_system_nanos = jvm_system_nanos;
 
+  return result;
+#elif defined(__FreeBSD__)
+  struct kinfo_proc *lproc;
+  size_t length;
+  int mib[3] = { CTL_KERN, KERN_PROC, KERN_PROC_ALL };
+  int result = cpu_load_total_process(psystemTotalLoad);
+
+  if (sysctl(mib, 3, NULL, &length, NULL, 0) == -1) {
+    return OS_ERR;
+  }
+
+  lproc = (struct kinfo_proc *)malloc(length);
+  if (!lproc) {
+    return OS_ERR;
+  }
+
+  if (sysctl(mib, 3, lproc, &length, NULL, 0) == -1) {
+    free(lproc);
+    return OS_ERR;
+  }
+
+  int active_processor_count = os::active_processor_count();
+  long jvm_user_nanos = 0;
+  long jvm_system_nanos = 0;
+
+  long total_cpu_nanos;
+  if(!now_in_nanos(&total_cpu_nanos)) {
```

Comment:
Note that I'm guessing this is because this is also inconsistent in the Apple version and this started with a copy and paste from there.  While there is substantial duplication between much of this and the Apple version, there is probably enough difference that it would be messier to avoid here than in the previous function.